### DISCUSSION TITLE
Issue with Combining Prometheus Alert Rules for Time Range Filtering

### DISCUSSION THREAD

**Discussion by YuHuai77** [prometheus/prometheus#15387] [Q&A]
### What did you do?

Hello, this is my syntax for querying data. I want to query data from 9am to 6pm every day, but these two statements can be separated to find some data. After merging them together, there is no data displayed, but the data does exist. Can you help me see what the problem is, or is it a syntax issue? thank you
`((node_filesystem_size_bytes{fstype!~"cifs|tmpfs"} - node_filesystem_free_bytes{fstype!~"cifs|tmpfs"})/ node_filesystem_size_bytes{fstype!~"cifs|tmpfs"} > bool 0.1 ==1 )   and ((hour(timestamp(up)) +8 >=bool 9==1 and hour(timestamp(up))+8 < bool 18==1))`

### What did you see instead? Under which circumstances?

Prometheus alarm rule limits time period

**Comment by roidelapluie** [comment]
Your query fails because the `and` operator requires both sides to match on all labels, but the filesystem and time-based metrics don’t share the same set of labels.

Here’s the correct query:
```promql
  (
          (
            node_filesystem_size_bytes{fstype!~"cifs|tmpfs"} - node_filesystem_free_bytes{fstype!~"cifs|tmpfs"}
          )
        /
          node_filesystem_size_bytes{fstype!~"cifs|tmpfs"}
      > bool
        0.1
    ==
      1
  )
and on ()
  ((hour(timestamp(up)) + 8 >= bool 9 == 1 and hour(timestamp(up)) + 8 < bool 18 == 1))
```

The correct query works because it uses `on()`, which ensures that the metrics are matched based only on some labels (in this case none). This allows combining the two expressions correctly without requiring a full set of matching labels.

**Comment by YuHuai77** [comment]
thank you
